src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -210,35 +240,22 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
     return null;
   }
 
-  /** Stream of clicks outside of the autocomplete panel. */
-  private get _outsideClickStream(): Observable<any> {
-    if (!this._document) {
-      return observableOf(null);
-    }
-
-    return RxChain.from(merge(
-      fromEvent(this._document, 'click'),
-      fromEvent(this._document, 'touchend')
-    )).call(filter, (event: MouseEvent | TouchEvent) => {
-      const clickTarget = event.target as HTMLElement;
-      const formField = this._formField ?
-          this._formField._elementRef.nativeElement : null;
-
-      return this._panelOpen &&
-             clickTarget !== this._element.nativeElement &&
-             (!formField || !formField.contains(clickTarget)) &&
-             (!!this._overlayRef && !this._overlayRef.overlayElement.contains(clickTarget));
-    }).result();
-  }
-
   /**
    * Sets the autocomplete's value. Part of the ControlValueAccessor interface
    * required to integrate with Angular's core forms API.
    *
    * @param value New value to be written to the model.
    */
   writeValue(value: any): void {
-    Promise.resolve(null).then(() => this._setTriggerValue(value));
+    // The stability check exists to make sure bindings have been initialized
+    // and avoid change detection errors.
+    if (this._zone.isStable) {
+      this._setTriggerValue(value);
+    } else {
+      this._nextStable.subscribe(() => {
+        this._setTriggerValue(value);
```

Comment:
Can you add a test that this still works the second time the value is set programmatically?

Reply:
It works in the demo, but I'll work on adding this test.

Reply:
Okay. I set the value multiple times in 'should update the display synchronously when set programmatically'